src/Donations/resources/admin/components/DonationDetailsPage/Tabs/Records/General/AssociatedDonor/index.tsx

```diff
@@ -0,0 +1,60 @@
+import { __ } from '@wordpress/i18n';
+import AdminSection, { AdminSectionField } from '@givewp/components/AdminDetailsPage/AdminSection';
+import { useFormContext } from 'react-hook-form';
+import { AsyncPaginate } from 'react-select-async-paginate';
+import styles from '../../styles.module.scss';
+import { useDonorAsyncSelect } from './useDonorAsyncSelect';
+
+/**
+ * @unreleased updated to async donor dropdown
+ * @since 4.6.0
+ */
+export default function AssociatedDonor() {
+    const { watch, setValue } = useFormContext();
+    const donationDonorId = watch('donorId');
+
+    const {
+        selectedOption,
+        loadOptions,
+        mapOptionsForMenu,
+        error,
+    } = useDonorAsyncSelect(donationDonorId || null);
+
+    const handleDonorChange = (selectedOption: any) => {
```

Comment:
I think we can improve this type by using something like this: `

```
const handleDonorChange = (selectedOption: DonorOption | null) => {
    setValue('donorId', selectedOption?.value || null, { shouldDirty: true });
};
```

Reply:
Resolved on a358bcc47bbf4decf619fb2a25de4f89dbf82d93